proto/minder/v1/minder.proto

```diff
@@ -4024,6 +4028,197 @@ message EntityInstance {
+        };
+
+        option (rpc_options) = {
+            target_resource: TARGET_RESOURCE_PROJECT
+            relation: RELATION_ENTITY_GET
+        };
+    }
+
+    // DeleteEntityById deletes an entity instance for a given entity ID
+    rpc DeleteEntityById (DeleteEntityByIdRequest) returns (DeleteEntityByIdResponse) {
+        option (google.api.http) = {
+            delete: "/api/v1/entity/id/{id}"
+        };
+
+        option (minder.v1.rpc_options) = {
+            target_resource: TARGET_RESOURCE_PROJECT
+            relation: RELATION_ENTITY_DELETE
+        };
+    }
+
+    // RegisterEntity creates a new entity instance
+    rpc RegisterEntity (RegisterEntityRequest) returns (RegisterEntityResponse) {
+        option (google.api.http) = {
+            post: "/api/v1/entity"
+            body: "*"
+        };
+
+        option (rpc_options) = {
+            target_resource: TARGET_RESOURCE_PROJECT
+            relation: RELATION_ENTITY_REGISTER
+        };
+    }
+}
+
+// ListEntitiesRequest is the request message for the ListEntities method
+message ListEntitiesRequest {
+    // context is the context in which the entities are listed
+    ContextV2 context = 1;
+
+    // entity_type is the type of entity to list
+    Entity entity_type = 2 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // cursor is the pagination cursor
+    Cursor cursor = 3;
+}
+
+// ListEntitiesResponse is the response message for the ListEntities method
+message ListEntitiesResponse {
+    // results is the list of entities
+    repeated EntityInstance results = 1 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // page is the pagination information
+    CursorPage page = 2;
+}
+
+// GetEntityByIdRequest is the request message for the GetEntityById method
+message GetEntityByIdRequest {
+    // context is the context in which the entity is evaluated
+    ContextV2 context = 1;
+
+    // id is the ID of the entity to get
+    string id = 2 [
+        (buf.validate.field).string = {uuid: true},
+        (google.api.field_behavior) = REQUIRED
+    ];
+}
+
+// GetEntityByIdResponse is the response message for the GetEntityById method
+message GetEntityByIdResponse {
+    // entity is the entity that was retrieved
+    EntityInstance entity = 1 [
+        (google.api.field_behavior) = REQUIRED
+    ];
+}
+
+// GetEntityByNameRequest is the request message for the GetEntityByName method
+message GetEntityByNameRequest {
+    // context is the context in which the entity is evaluated
+    ContextV2 context = 1;
+
+    // name is the name of the entity to get
+    string name = 2 [
+        (buf.validate.field).string = {
+            pattern: "^[A-Za-z][-/[:word:]]*$",
+            max_len: 200,
+        },
+        (google.api.field_behavior) = REQUIRED
+    ];
+
+    // entity_type is the type of entity to get
+    Entity entity_type = 3 [
+        (google.api.field_behavior) = REQUIRED
+    ];
```

Comment:
Should the URL for this be `/api/v1/entity/{type}/{name=**}`?

I think it's fine to leave this like this for now, and possibly change the URL later

Reply:
I like this!